### ISSUE TITLE
No 'Play Store' gradle tasks appearing

### ISSUE THREAD

**Issue by scottyab** [Triple-T/gradle-play-publisher#250]
I have configured the play-publisher plugin on an Android project as per readme but none of the play publish gradle tasks appear (i.e `publishApkRelease`). I have a several other similar projects configured and play-publisher tasks are available. This project is SDK project so wondered if there was a conflict with maven publisher plugin? I've tried to isolate by removing other gradle plugins that might be causing conflict but still no joy. 

I noticed there's no validation when the `pk12File` property is an invalid file (but that doesn't stop the `publishApkRelease` task from being available). So looking for suggestions to diagnose why this might be happening? Plugin/config error? Is there a way of enabling a verbose log for play-publisher plugin specifically? 

It's great plugin! thanks for the efforts and any advice offered.

**Comment by asinel** [comment]
Same problem

**Comment by asinel** [comment]
Solved it with adding signingConfig

**Comment by bhurling** [comment]
@scottyab does your SDK project apply the `com.android.application` plugin? Sounds like you're actually building a library which does not produce an APK for upload.

**Comment by scottyab** [comment]
Yes. the app/.apk that I'm trying to publish uses `com.android.application` already tried adding signing config but still no joy. 

We have the following modules in the project: 

* /app - com.android.application plugin apk that I'm trying to publish to play store 
* /LibCore - com.android.library plugin, aar that we publish to bintray
* /LibUi - com.android.library plugin, aar that we publish to bintray

Thanks for the response.

**Comment by bhurling** [comment]
Any chance your sample project is on github?

**Comment by SUPERCILEX** [comment]
Closing due to lack of response.

**Comment by carstenhag** [comment]
Hey, I think I'm having the same (or at least a similar) issue.

I am trying to automatically deploy my app via gitlab CI and this plugin. On my machine & AS, I can see the Publishing tasks with `gradlew tasks`.

On the CI server, however, no Publishing tasks are found.

My code is at https://gitlab.com/carstenhag/android-mosbach-map and these are the CI logs:

```
Accept? (y/N): All SDK package licenses accepted

$ set -o pipefail
$ echo $KEYS_JSON | base64 -di > playstoreapi.json
$ echo $SIGNING_KEY_JKS | base64 -di > keystore.jks
$ echo $SIGN_GRADLE | base64 -di > sign.gradle
$ ./gradlew tasks
Downloading https://services.gradle.org/distributions/gradle-4.10.1-all.zip
...............................................................................................................
Unzipping /root/.gradle/wrapper/dists/gradle-4.10.1-all/455itskqi2qtf0v2sja68alqd/gradle-4.10.1-all.zip to /root/.gradle/wrapper/dists/gradle-4.10.1-all/455itskqi2qtf0v2sja68alqd
Set executable permissions for: /root/.gradle/wrapper/dists/gradle-4.10.1-all/455itskqi2qtf0v2sja68alqd/gradle-4.10.1/bin/gradle

Welcome to Gradle 4.10.1!

Here are the highlights of this release:
 - Incremental Java compilation by default
 - Periodic Gradle caches cleanup
 - Gradle Kotlin DSL 1.0-RC6
 - Nested included builds
 - SNAPSHOT plugin versions in the `plugins {}` block

For more details see https://docs.gradle.org/4.10.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

> Task :tasks

------------------------------------------------------------
All tasks runnable from root project
------------------------------------------------------------

Android tasks
-------------
androidDependencies - Displays the Android dependencies of the project.
signingReport - Displays the signing info for the base and test modules
sourceSets - Prints out all the source sets defined in this project.

Build tasks
-----------
assemble - Assemble main outputs for all the variants.
assembleAndroidTest - Assembles all the Test applications.
build - Assembles and tests this project.
buildDependents - Assembles and tests this project and all projects that depend on it.
buildNeeded - Assembles and tests this project and all projects it depends on.
bundle - Assemble bundles for all the variants.
clean - Deletes the build directory.
cleanBuildCache - Deletes the build cache directory.
compileDebugAndroidTestSources
compileDebugSources
compileDebugUnitTestSources
compileReleaseSources
compileReleaseUnitTestSources

Build Setup tasks
-----------------
init - Initializes a new Gradle build.
wrapper - Generates Gradle wrapper files.

Cleanup tasks
-------------
lintFix - Runs lint on all variants and applies any safe suggestions to the source code.

Help tasks
----------
buildEnvironment - Displays all buildscript dependencies declared in root project 'android-mosbach-map'.
components - Displays the components produced by root project 'android-mosbach-map'. [incubating]
dependencies - Displays all dependencies declared in root project 'android-mosbach-map'.
dependencyInsight - Displays the insight into a specific dependency in root project 'android-mosbach-map'.
dependentComponents - Displays the dependent components of components in root project 'android-mosbach-map'. [incubating]
help - Displays a help message.
model - Displays the configuration model of root project 'android-mosbach-map'. [incubating]
projects - Displays the sub-projects of root project 'android-mosbach-map'.
properties - Displays the properties of root project 'android-mosbach-map'.
tasks - Displays the tasks runnable from root project 'android-mosbach-map' (some of the displayed tasks may belong to subprojects).

Install tasks
-------------
installDebug - Installs the Debug build.
installDebugAndroidTest - Installs the android (on device) tests for the Debug build.
uninstallAll - Uninstall all applications.
uninstallDebug - Uninstalls the Debug build.
uninstallDebugAndroidTest - Uninstalls the android (on device) tests for the Debug build.
uninstallRelease - Uninstalls the Release build.

Verification tasks
------------------
check - Runs all checks.
connectedAndroidTest - Installs and runs instrumentation tests for all flavors on connected devices.
connectedCheck - Runs all device checks on currently connected devices.
connectedDebugAndroidTest - Installs and runs the tests for debug on connected devices.
deviceAndroidTest - Installs and runs instrumentation tests using all Device Providers.
deviceCheck - Runs all device checks using Device Providers and Test Servers.
lint - Runs lint on all variants.
lintDebug - Runs lint on the Debug build.
lintRelease - Runs lint on the Release build.
lintVitalRelease - Runs lint on just the fatal issues in the release build.
test - Run unit tests for all variants.
testDebugUnitTest - Run unit tests for the debug build.
testReleaseUnitTest - Run unit tests for the release build.

To see all tasks and more detail, run gradlew tasks --all

To see more detail about a task, run gradlew help --task <task>

BUILD SUCCESSFUL in 46s
1 actionable task: 1 executed
$ ./gradlew publishBundle --no-daemon
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/4.10.1/userguide/gradle_daemon.html.
Daemon will be stopped at the end of the build stopping after processing

FAILURE: Build failed with an exception.

* What went wrong:
Task 'publishBundle' not found in root project 'android-mosbach-map'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 14s
```

**Comment by SUPERCILEX** [comment]
1. You have to [provide credentials](https://github.com/Triple-T/gradle-play-publisher#encrypting-service-account-keys) if you want stuff to work
1. If you don't need to deploy from CI, this has been fixed: #461

**Comment by carstenhag** [comment]
Thanks. My gradle file wasn't including the other gradle file with the signing config.
